## Utilizando Web-Scraping para levantar a literatura

#### Passo 1: instalar bibliotecas

In [14]:
pip install beautifulsoup4


Note: you may need to restart the kernel to use updated packages.


#### Passo 2: Importar bibliotecas

In [4]:
import requests
from bs4 import BeautifulSoup

#### Passo 3: Formatar link de busca

In [16]:
# "URL_BUSCA é onde devemos apenas buscar a url"
# Copie o modelo do link de busca
url_busca = 'https://www.periodicos.capes.gov.br/index.php/acervo/buscador.html?q='


# Informe a palvra chave
palavra_chave = 'everson zoio'


# Formata link
link = url_busca+palavra_chave
print(link)

https://www.periodicos.capes.gov.br/index.php/acervo/buscador.html?q=everson zoio


#### Passo 4: Obter informações do link

In [17]:
# Requisita informações do site
response= requests.get(link)

# Verifica status de acesso
print(response.status_code)


403


Esse **403 Forbidden** acontece porque o Portal da Capes possui mecanismos de proteção contra bots e scrapers simples, e o requests.get() sem nenhum cabeçalho já denuncia que não é um navegador “de verdade”.

#### 4.1 Usar um User-Agent válido

Alguns sites **bloqueiam requisições que não parecem vir de navegadores**. Então você pode **enviar cabeçalhos simulando um browser**.

In [19]:
# Cabeçalho com User-Agent
headers = {"user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/151.0.0.0 Safari/537.36"}


# Requisita informações do site
response = requests.get(link, headers=headers)


# Verifica status de acesso
print(response.status_code)

403


#### Passo 5: Encontrar elementos

In [ ]:
# Encontrar por id


In [ ]:
#Título


#### Passo 6: Armazenar dados

In [ ]:
# Cria uma lista vazia para guardar dados


# Cria laço de repetição para filtrar todos os dados


#### Passo 7: Visualizar listas

#### Passo 8: Organizar tabelas e salvar

In [ ]:
# Importar biblioteca Pandas para criar DataFrames


In [ ]:
# Transforma arrays em DataFrame


# Salvar DataFrame


In [ ]:
# Visualizar DataFrame


#### Exercício: Extrair as demais informações abaixo
- Título
- Autores
- Editora
- Volume
- Edição (Número)
- Idioma
- Ano
- Tópicos
- DOI
- ISSN
- Resumo (sobre)
- Link

#### Tentativa de resposta